### PR TITLE
fix(developer): server download Keyman link

### LINKED ISSUE #9702: bug(developer): The "Download and Install Keyman" option in the menu list within the localhost:8008 browser window is unresponsive or not functioning
### Describe the bug

I observed that when I click the "Download and Install Keyman" option from the menu list, it does not trigger any action on the screen.

### Reproduce the bug

1. Install Keyman Developer 17.0.186-alpha build.
2. Open a web keyboard in a localhost:8008 browser using Keyman Developer.
3. Click the top left icon (list menu) from the browser. 
4. Click the 'Download and install Keyman' option. 

I noticed that after clicking the 'Download and install Keyman' option, it does not trigger any action on the screen. 

I have attached the Screenshot for reference. 

   

### Expected behavior

Selecting the 'download and install Keyman' option should trigger the opening of the Keyman download and installation window.

### Related issues

_No response_

### Keyman apps

- [ ] Keyman for Android
- [ ] Keyman for iPhone and iPad
- [ ] Keyman for Linux
- [ ] Keyman for macOS
- [ ] Keyman for Windows
- [X] Keyman Developer
- [ ] KeymanWeb
- [ ] Other - give details at bottom of form

### Keyman version

17.0.186-alpha

### Operating system

Windows 10 OS, Windows 11 OS (VM)

### Device

_No response_

### Target application

_No response_

### Browser

_No response_

### Keyboard name

_No response_

### Keyboard version

_No response_

### Language name

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #9702.

# User Testing

Setup: you will need to have Keyman Developer Server running on your Windows computer for all these tests, in the same way as described in #9702. Then, you will need to locate the appropriate IP address on your local network for your computer, and ensure that the test devices are on the same local network. For example, on my machine, the address I use is http://192.168.129.109:8008:

image

Then, open that page on the target device and test the Download Keyman link there.

GROUP_WINDOWS: Test on Windows
GROUP_MAC: Test on MacOS
GROUP_ANDROID: Test on MacOS
GROUP_IOS: Test on iOS
GROUP_LINUX: Test on Linux

TEST_LINK: Test ths the Download Keyman link starts a download for Keyman or loads a page appropriate for the device

### CODE REVIEW

**Review by rc-swag — APPROVED** [review]
lgtm

File: developer/src/server/src/site/packages.js [hunk 1]
```diff
@@ -11,12 +11,12 @@ menuDropdown.onclick = (value) => {
   menuDropdown.set(''); // we never show an 'active' package
   if(value == '#install-keyman') {
     let href = '';
-    switch(keyman.util.device.OS) {
-      case 'iOS':     href = 'https://keyman.com/go/developer/'+versionMajor+'/ios-app'; break;
-      case 'Android': href = 'https://keyman.com/go/developer/'+versionMajor+'/android-app'; break;
-      case 'Linux':   href = 'https://keyman.com/linux/download'; break;
-      case 'Windows': href = 'https://keyman.com/go/download/keyman-windows'; break;
-      case 'MacOSX':  href = 'https://keyman.com/go/download/keyman-mac'; break;
+    switch(keyman.config.hostDevice.OS) {
```
Comment (mcdurdin) [inline developer/src/server/src/site/packages.js:14]:
```suggestion
    switch(keyman.config.hostDevice.OS) {  // Note: KeymanWeb internal API
```

**Comment by bharanidharanj** [comment]
# Test Results

### GROUP_WINDOWS: Test on Windows

* **TEST_LINK (FAILED):** Tested with Keyman Developer 17.0.196-alpha-test-9822 (downloaded it from TeamCity) in both Windows 10 (Native) OS and here is my observation: 1. In Windows 10 OS, I got a warning message 'Keyman Developer Server has not yet started successfully. Keep waiting?". Clicking the OK button repeats this message on the screen.

**Comment by mcdurdin** [comment]
@keymanapp-test-bot retest

**Comment by bharanidharanj** [comment]
> > * Tested with Keyman Developer 17.0.196-alpha-test-9822 (downloaded it from TeamCity) in both Windows 10 (Native) OS and here is my observation: 1. In Windows 10 OS, I got a warning message 'Keyman Developer Server has not yet started successfully. Keep waiting?". Clicking the OK button repeats this message on the screen.
> 
> @bharanidharanj I tried but was unable to reproduce this problem. Can you re-test this, and look to see if the Keyman Developer Server has started down in the system notification area?

Sure @mcdurdin . I will re-test it and will post my result.

**Comment by bharanidharanj** [comment]
# Test Results

### GROUP_WINDOWS: Test on Windows

* **TEST_LINK (PASSED):** Tested with the attached PR build (Keyman Developer 17.0.196-alpha-test-9822) on Windows 11 OS (VM) and here is my observation: 1. Able to see the Keyman Server is running in the background. (the icon is displayed in the taskbar). 2. Opened the test page in the local browser on the target device (Windows) then clicked the Download Keyman link there. 3. Verified that clicking the Download link button would initiate the download of the Keyman stable build.

..Opened the web page in the given web address link

   


..Download link working in Windows target device

   


### GROUP_MAC: Test on MacOS

* **TEST_LINK (PASSED):** 1. Opened the test page in the local browser on the target device (macOS) and then clicked the Download Keyman link there. 2. Verified that clicking the Download link button would initiate the download of the Keyman stable build.

   


### GROUP_ANDROID: Test on Android

* **TEST_LINK (PASSED):** 1. Opened the test page in the local browser on the target device (Nexus 9) then clicked the Download Keyman link there. 2. Verified that clicking the Download link button would initiate the download of the Keyman stable build.

   


### GROUP_IOS: Test on iOS

* **TEST_LINK (PASSED):** 1. Opened the test page in the local browser on the target device (iPhone 5S) then clicked the Download Keyman link there. 2. Verified that clicking the Download link button would initiate the download of the Keyman stable build.

   


### GROUP_LINUX: Test on Linux

* **TEST_LINK (PASSED):** 1. Opened the test page in the local browser on the target device (Linux) then clicked the Download Keyman link there. 2. Verified that clicking the Download link button would initiate the download of the Keyman stable build.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.199-alpha](https://keyman.com/downloads/releases/alpha/17.0.199)

**Pull request merged**